internal/services/cosmos/cosmosdb_account_resource.go

```diff
@@ -1706,3 +1784,47 @@ func flattenCosmosdbAccountDatabasesToRestore(input *[]documentdb.DatabaseRestor
 
 	return results
 }
+
+func checkCapabilitiesCanBeUpdated(kind string, oldCapabilitiesMap map[string]interface{}, newCapabilitiesMap map[string]interface{}) error {
+	// The feedback from service team : "DisableRateLimitingResponses", "AllowSelfServeUpgradeToMongo36","EnableAggregationPipeline","MongoDBv3.4"
+	// and "mongoEnableDocLevelTTL" of capabilities can be added to an existing account, others can not.
+	canBeAddedCapsList := []string{string(databaseAccountCapabilitiesDisableRateLimitingResponses), string(databaseAccountCapabilitiesAllowSelfServeUpgradeToMongo36),
+		string(databaseAccountCapabilitiesEnableAggregationPipeline), string(databaseAccountCapabilitiesMongoDBv34), string(databaseAccountCapabilitiesMongoEnableDocLevelTTL)}
+
+	canBeAddedCapsMap := make(map[string]interface{})
+	for _, v := range canBeAddedCapsList {
+		canBeAddedCapsMap[strings.ToLower(v)] = capabilitiesToKindMap[strings.ToLower(v)]
+	}
+
+	// The feedback from service team : only "DisableRateLimitingResponses" of capabilities can be removed to an existing account.
+	canBeRemovedCaps := string(databaseAccountCapabilitiesDisableRateLimitingResponses)
+
+	for nck, ncv := range newCapabilitiesMap {
+		if _, ok := oldCapabilitiesMap[nck]; !ok {
+			v, ok := canBeAddedCapsMap[nck]
+			if !ok {
+				return fmt.Errorf("unable to add the `%s` to an existing account. Possible values that can be added are %s", ncv, canBeAddedCapsList)
+			}
+			canBeAdded := false
+			for _, kind := range v.([]string) {
+				if kind == strings.ToLower(kind) {
+					canBeAdded = true
+					break
+				}
+			}
+			if !canBeAdded {
+				return fmt.Errorf("unable to add the `%s` to an existing account of kind is `%s`", ncv, kind)
+			}
+		}
+	}
+
+	for ock, ocv := range oldCapabilitiesMap {
+		if _, ok := newCapabilitiesMap[ock]; !ok {
+			if !strings.EqualFold(ock, canBeRemovedCaps) {
+				return fmt.Errorf("unable to remove the `%s` from an existing account. Possible value that can be removed is `%s`", ocv, canBeRemovedCaps)
+			}
+		}
+	}
```

Comment:
```suggestion
	// first check the new capabilities can be added
	for _, capability := range *newCapabilities {
		if capability.Name == nil {
			continue
		}
		existedPreviously := false
		for _, existing := range *oldCapabilities {
			if existing.Name != nil && strings.EqualsFold(*existing.Name, *capability.Name) {
				existedPreviously = true
				break
			}
		}
		if !existedPreviously {
			continue
		}
		
		// retrieve a list of capabilities for this DB type
		supportedKindsForCapability ok := capabilitiesToKindMap[strings.ToLower(capability)]
		if !ok {
			return fmt.Errorf("internal-error: missing mapping between Kind %q and Capabilities for %q", kind, capability)
		}
		
		// first check if this is supported
		if _, isSupported := supportedKindsForCapability[strings.ToLower(kind)]; !isSupported {
			return fmt.Errorf("the Capability %q was not supported for the Kind %q - only the following Kinds support updating this Capability: %+v", capability, kind, strings.Join(supportedKindsForCapability " / "))
		}
		
		// then check if it can be added via an update
		if !utils.SliceContainsValue(canBeAddedCaps, strings.ToLower(capability)) {
			return fmt.Errorf("the Capability %q for the Kind %q cannot be updated it must be set at creation", capability, kind)
		}
	}

	// then check if we're removing any that they can be removed
	for capability := range *oldCapabilities {
		existsNow := false
		for _, new := range *newCapabilities {
			if new.Name != nil && strings.EqualsFold(*new.Name, *capability.Name) {
				existsNow = true
				break
			}
		}
		if existsNow {
			continue
		}
		
		if !utils.SliceContainsValue(canBeRemovedCaps, strings.ToLower(capability)) {
			return fmt.Errorf("the Capability %q for the Kind %q cannot be removed it must be omitted at creation", capability, kind)
		}
	}
```

Reply:
Fixed.